code/game/objects/machinery/doors/multi_tile.dm

```diff
@@ -103,6 +106,149 @@
 /obj/machinery/door/airlock/multi_tile/mainship/generic/rnr
 	name = "\improper Rest and Recreation"
 
+/obj/machinery/door/airlock/multi_tile/mainship/generic2
+	name = "\improper Airlock"
+	icon = 'icons/obj/doors/mainship/2x1almayerdoor.dmi'
+	opacity = FALSE
+	glass = TRUE
+
+/obj/machinery/door/airlock/multi_tile/mainship/generic2/glass
+	name = "\improper Glass Airlock"
+	icon = 'icons/obj/doors/mainship/2x1almayerdoor_glass.dmi'
+
+//PREP DOORS
+
+/obj/machinery/door/airlock/multi_tile/mainship/marine
+	name = "\improper Squad Preparations"
+	icon = 'icons/obj/doors/mainship/2x1prepdoor.dmi'
+	req_access = list(ACCESS_MARINE_PREP)
+	opacity = FALSE
+	glass = TRUE
+
+/obj/machinery/door/airlock/multi_tile/mainship/marine/requisitions
+	name = "\improper Requisitions Bay"
+	req_one_access = list(ACCESS_MARINE_LOGISTICS, ACCESS_MARINE_CARGO)
+
```

Comment:
you dont even set the icon/iconstates here go do it properly im not reviewing this